### ISSUE TITLE
Articulate 5.0.1 causes all node URLs except the homepage to return 404 not found

### ISSUE THREAD

**Issue by SRUC** [Shazwazza/Articulate#432]
Articulate 5.0.1 causes all node URLs except the homepage to return 404 not found.

Tested using Umbraco.TheStarterKit and OskiStarterKit in Umbraco v12.3.3 and v13.0.0 RC2

After removing Articulate 5.0.1 and rebuilding the solution, node URLs function as expected.

**Comment by Shazwazza** [comment]
What root nodes do you have then with all of these starter kits? Do you have multiple root nodes? Do you have domains defined for them? There can only be a single root node without a domain assigned in Umbraco.

**Comment by gavinfaux** [comment]
Have seen same issue on a previous working 5.0.0 site, two root nodes, first node has domain / culture assigned, second node nothing assigned (container for content blocks and settings). Uninstall of 5.0.1 and reverting to 5.0.0 allows rest of site to render. Have not looked into further as yet.

**Comment by khraibani** [comment]
Same issue happening with version 5.0.1 as @gavinfaux

**Comment by Shazwazza** [comment]
@khraibani exactly the same? 2 root nodes, one domain assigned, or something different?

**Comment by MartinAmsinck** [comment]
I ran in to something similar i think.
I'm on a 13.0.3, and i'm using a fork of Articulate, not an installed package.

I have multiple Sites, under one rootnode, and each of them has its own domain
<img width="142" alt="image" src="https://github.com/Shazwazza/Articulate/assets/1222544/76f29fad-7c52-452b-86ae-60497cb32d9d">

But nothing but the first rootnode gets hit (the "Sites" node), everything else is a 404.
The "Sites" node has no domain.

When removing the Articulate code, everything works as expected.

It seems that when hitting this part of the code in `ArticulateRouteValueTransformer.ShouldCheck()`, `umbracoRouteValues?.PublishedRequest` is null

<img width="461" alt="image" src="https://github.com/Shazwazza/Articulate/assets/1222544/44e95e20-782a-40bb-872a-d720f0c8ea2f">

The comment `Ensure it runs AFTER Umbraco ` made me bump the value from 100 to 1000. 
Seems to do the trick for me. 
<img width="636" alt="image" src="https://github.com/Shazwazza/Articulate/assets/1222544/24a00293-d717-446f-abe4-daf3aa49164f">

Hope this helps

**Comment by nvisage-gf** [comment]
@MartinAmsinck Thanks for suggested fix, our site working fine now, PR submitted.

Unsure how to debug/list route order to verify if 1000 is optimal.

**Comment by khraibani** [comment]
@Shazwazza I was running 
- Umbraco v12.3.6 
- Articulate 5.0.1 
- 1 root node
I reverted back to version 5.0.0 and its working apart from the categories/tags I will raise that as another issue

**Comment by Shazwazza** [comment]
@khraibani are you running in Umbraco Cloud? I tweeted about this a while back https://twitter.com/Shazwazza/status/1720142890901684630 and had to ship 5.0.1 to make it work for the virtual routes.

**Comment by Shazwazza** [comment]
Though, maybe my workarounds (see https://github.com/Shazwazza/Articulate/commit/6d0b672569cc6a7e100575058ff80cb19be566ec, https://github.com/Shazwazza/Articulate/commit/e3d34ba14e5865b866e949425c20058bf73431ea) was resolved by changing the MapDynamicControllerRoute to 1000, not sure

**Comment by khraibani** [comment]
@Shazwazza no its a local version of Umbraco (self hosted) should that make a difference?

**Comment by eqtr-ab** [comment]
Hey there, we encountered this issue today while adding v5.0.1 of Articulate to an Umbraco Cloud site running v13.0.3 of the CMS. Using v5.0.0 instead resolves the issue.

**Comment by BarryGibson** [comment]
I too had the same issue. I found that the pages were available if I added // after the base domain i.e. https://localhost:44372//Page-name.
This was fixed in the latest Dev version as well as using V5.0.0.

**Notes to help debug if needed:**
It only started happening after I added a custom SurfaceController. 
I confirmed this by creating a clean Umbraco 13.1 build
The site worked correctly with Articulate V5.0.1 without a SurfaceController and with a SurfaceController but not Articulate but with both it returned "404 not found" - the 404 wasn't an Umbraco handled 404 it was a system 404.
For testing the SurfaceController was a pretty empty one as shown below.

```
using Microsoft.AspNetCore.Mvc;
using Umbraco.Cms.Core.Cache;
using Umbraco.Cms.Core.Logging;
using Umbraco.Cms.Core.Routing;
using Umbraco.Cms.Core.Services;
using Umbraco.Cms.Core.Web;
using Umbraco.Cms.Infrastructure.Persistence;
using Umbraco.Cms.Web.Website.Controllers;

namespace UmbracoProject1.Controllers
{
	public class SampleController : SurfaceController
	{
		public SampleController(IUmbracoContextAccessor umbracoContextAccessor, IUmbracoDatabaseFactory databaseFactory, ServiceContext services, AppCaches appCaches, IProfilingLogger profilingLogger, IPublishedUrlProvider publishedUrlProvider) : base(umbracoContextAccessor, databaseFactory, services, appCaches, profilingLogger, publishedUrlProvider)
		{
		}

		public IActionResult Index()
		{
			return Content("Hello world");
		}
	}
}
```

**Comment by JamesDodwell** [comment]
I've  encountered this issue too and the // after the base url does mean the pages load e.g. https://localhost:44310//articles/latte-art/

**Comment by westinlennoxbrandner** [comment]
We are seeing this same issue with 5.0.1. Our solution was to downgrade to 5.0.0 which has fixed the issue.

**Comment by The-algar** [comment]
Had the same issue Umbraco CMS 13.2.2, Articulate latest version, One root node
once Articulate installed, 404 error for all pages
Question: How to Setup An Articulate Blog Package to An Existing Umbraco Layout 
Is there's a process on how to do this?

**Comment by Shazwazza** [comment]
really sorry for the accidental delay in getting these shipped as part of releases, I though this was already done 🤦‍♂️🤦‍♂️ 5.0.2 and 5.0.3 are now shipped.

I'm not sure if those 'fix' this issue but hopefully will help some folks, I still need to replicate locally

**Comment by BarryGibson** [comment]
> really sorry for the accidental delay in getting these shipped as part of releases, I though this was already done 🤦‍♂️🤦‍♂️ 5.0.2 and 5.0.3 are now shipped.
> 
> I'm not sure if those 'fix' this issue but hopefully will help some folks, I still need to replicate locally

Thanks this is now fixed when using 5.0.3
